### PR TITLE
Cleaned docker files

### PR DESCRIPTION
**Pull request by harshitsharma2801**
1) Cleaned docker files for both frontend and backend to have separate build and runtime stage.
2)Bumped carbon
3)The logic to increase the number of tickets in point-to-point and req-res had a small bug , fixed that.

### CODE REVIEW

**Comment by harshitsharma2801** [comment]
Fixed the Pub/Sub UI to display the required components in the sidebar.

File: ibm-messaging-mq-cloud-showcase-app/backend/Dockerfile [hunk 1]
```diff
@@ -12,28 +12,87 @@
...
 
+# This Dockerfile has two separate stages.
+#The first stage is builds the Node program.
+# The second stage is a runtime-only container that holds just the things we need to
+# execute the compiled program.
+
+###########################################################
+# This starts the BUILD phase
+###########################################################
 FROM ubuntu:20.04 as builder
```
Comment (chughts) [inline ibm-messaging-mq-cloud-showcase-app/backend/Dockerfile:23]: As we are building a node.js image, is there a node.js base image that would the need to install node etc. ?

File: ibm-messaging-mq-cloud-showcase-app/frontend/AWS-MQ-showcase/package.json [hunk 2]
```diff
@@ -12,14 +12,14 @@
     "format": "prettier --write \"**/*.{js,md,scss}\"",
     "format:diff": "prettier --list-different \"**/*.{js,md,scss}\"",
     "format:staged": "prettier --write",
-    "start": "react-scripts start",
+    "start": "BROWSER=none react-scripts start",
```
Comment (chughts) [inline ibm-messaging-mq-cloud-showcase-app/frontend/AWS-MQ-showcase/package.json:15]: Need to add explanation of why this is necessary for WSL.

File: ibm-messaging-mq-cloud-showcase-app/frontend/nginx.conf [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+    location / {
+        root   /usr/share/nginx/html;
+        index  index.html index.htm;
+        try_files $uri $uri/ /index.html;
+    }
+
+    #Set up reverse proxy to redirect requests to backend
+    location /api/ {
+        proxy_pass http://be:8080/api/;
```
Comment (chughts) [inline ibm-messaging-mq-cloud-showcase-app/frontend/nginx.conf:29]: Http / Https, Host and port need to be set via the environment variables:

````
const be = process.env.REACT_APP_BE_HOST;
const be_port = process.env.REACT_APP_BE_PORT || "8080";

const be_tls = process.env.REACT_APP_BE_TLS || false;
const HTTP_PROTOCOL = be_tls ? "https://" : "http://";
````

See existing proxy setup : https://github.com/ibm-messaging/mq-dev-patterns/blob/2b66777a8208e149bea3c3e24da435fa7b14ae58/ibm-messaging-mq-cloud-showcase-app/frontend/AWS-MQ-showcase/src/setupProxy.js#L19

**Review by chughts — CHANGES_REQUESTED** [review]

File: ibm-messaging-mq-cloud-showcase-app/frontend/AWS-MQ-showcase/src/setupProxy.js [hunk 4]
```diff
@@ -17,8 +17,8 @@
 const { createProxyMiddleware } = require('http-proxy-middleware');
```
Comment (chughts) [inline ibm-messaging-mq-cloud-showcase-app/frontend/AWS-MQ-showcase/src/setupProxy.js:17]: Update copyright date

File: ibm-messaging-mq-cloud-showcase-app/frontend/Dockerfile [hunk 5]
```diff
@@ -1,5 +1,5 @@
 
-#  Copyright 2022, 2023 IBM Corp.
+#  Copyright 2022, 2023, 2024 IBM Corp.
```
Comment (chughts) [inline ibm-messaging-mq-cloud-showcase-app/frontend/Dockerfile:2]: fix copyright date

File: ibm-messaging-mq-cloud-showcase-app/frontend/nginx.conf (hunk 3 above)
Comment (chughts) [reply ibm-messaging-mq-cloud-showcase-app/frontend/nginx.conf:29]: should this code be using `be_port` in stead of 8080 ?

**Review by chughts — CHANGES_REQUESTED** [review]

File: ibm-messaging-mq-cloud-showcase-app/backend/Dockerfile [hunk 6]
```diff
@@ -20,8 +19,7 @@
 ###########################################################
 # This starts the BUILD phase
 ###########################################################
-FROM node:latest as builder
-
+FROM node:20.9.0 as builder
```
Comment (chughts) [inline ibm-messaging-mq-cloud-showcase-app/backend/Dockerfile:22]: The build is at 20.9.0, yet the runtime is at 18-slim, these should be aligned.

File: ibm-messaging-mq-cloud-showcase-app/frontend/Dockerfile [hunk 7]
```diff
@@ -1,5 +1,5 @@
 
-#  Copyright 2022, 2023, 2024 IBM Corp.
+#  Copyright 2022 , 2024 IBM Corp.
```
Comment (chughts) [inline ibm-messaging-mq-cloud-showcase-app/frontend/Dockerfile:2]: remove space between 2022 and ,

**Review by chughts — CHANGES_REQUESTED** [review]
Only update the copyright on files that have changed. Many of the files in this pull request are showing the only change as the copyright line. 

If a file has been updated the format should be 

```Copyright 2022, 2024 IBM Corp.````

Note the spacing. 1st date is when the file was initially created, 2nd date is when the last modifications were made to the file.

**Review by chughts — APPROVED** [review]

**Pull request merged**